src/main/scala/mesosphere/marathon/core/base/CurrentRuntime.scala

```diff
@@ -0,0 +1,37 @@
+package mesosphere.marathon.core.base
+
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ ExecutionContext, TimeoutException, Await, Future }
+import scala.util.control.NonFatal
+import scala.concurrent.duration._
+
+object CurrentRuntime {
+
+  private[this] val log = LoggerFactory.getLogger(getClass.getName)
+
+  /**
+    * Exit this process in an async fashion.
+    * First try exit regularly in the given timeout. If this does not exit in time, we halt the system.
+    *
+    * @param exitCode the exit code to signal.
+    * @param waitForExit the time to wait for a normal exit.
+    * @return the Future of this operation.
+    */
+  //scalastyle:off magic.number
+  def asyncExit(exitCode: Int = 9,
+                waitForExit: FiniteDuration = 10.seconds)(implicit ec: ExecutionContext): Future[Unit] = {
+    Future({
+      try {
```

Comment:
Add blocking{} around block Future blocks to inform ExecutionContext. (scala.concurrent.blocking).

Otherwise seems reasonable. I think it'd be even better to use a timer of sorts, eg:
'def timeout[T](in: Duration)(f: => T): Future[T]' and 'def timeout[T](in: Duration)(f: => Future[T]): Future[T]'

That way we can quickly express "operation X should complete within N" for any arbitrary method.

The method taking in f: => T should wrap 'f' in blocking()